### ISSUE TITLE
RTDB Sync doesn't return if another sync with same ref exists

### ISSUE THREAD

**Issue by luskin** [n6g7/redux-saga-firebase#186]
There seems to be an issue where `redux-saga-firebase` will not dispatch the `successActionCreator` nor throw an error nor dispatch a `failureActionCreator` when trying to put a listener on a ref that is already being listened to elsewhere in the instance. Albeit this is an edge case, but there is a certain flow a user could take in our application where there would need to be two separate ref listeners for the same path. The standard firebase `.on` and `.once` work fine with multiple subscribers to the same path so this package should as well.

```
  const customerOrganizationRef = Firebase.db.ref(`organization/${payload}`)

  // This code returns fine
  customerOrganizationRef.once('value', snap => {
    console.log('The snap value: ', snap.val())
  })

  // This code will not dispatch the success action, and if a failure action
  // is included it does not return either
  const organizationTask = yield fork(Firebase.saga.database.sync, customerOrganizationRef, {
    successActionCreator: sync.UPDATE
  })

  yield take('LOGOUT')
  yield cancel(organizationTask)
```

Without diving into the code of this package, my assumption is that the package key's the refs and notices the key exists and therefore does not follow through with creating the sync, @n6g7 would I be correct in this assumption?

**Comment by n6g7** [comment]
That seems weird, there's no caching or refs in this library (see the [syncChannel function](https://github.com/n6g7/redux-saga-firebase/blob/d7f5ce4ff73a93c0e139da13789c4c85dfbc96c2/src/utils.js#L3-L23) where the corresponding logic is implemented).

I tried to [reproduce this bug in the example site](https://github.com/n6g7/redux-saga-firebase/blob/lib/issue-186/example/src/redux/sagas/todos/realtime.js#L30-L39) but I saw the success action being dispatched without issue.

Two questions to try to narrow this down:
- Does the same thing happen with firestore?
- Could you provide a more complete reproducible example or advise on what's missing in my branch?

Thanks for the feedback!

**Comment by luskin** [comment]
I'll try and find some time to create a reproducible example in the coming week.

**Comment by luskin** [comment]
Additionally using the channel method works:

This Does Not Work:
```
  const customerOrganizationUsersRef = Firebase.db
    .ref('user')
    .orderByChild('organization')
    .equalTo(payload)
  usersTask = fork(Firebase.saga.database.sync, customerOrganizationUsersRef, {
    successActionCreator: syncUsers.UPDATE,
    failureActionCreator: syncUsers.FAILURE
  })
```

This Does Work:
```
  const channel = yield call(Firebase.saga.database.channel, customerOrganizationUsersRef)
  while (true) {
    const { value } = yield take(channel)
    yield put(syncUsers.UPDATE(value))
  }
```

This leads me to believe there is a bug somewhere with the `.sync` functionality

**Comment by luskin** [comment]
@n6g7 I found the problem. If there is already an existing sync on a particular reference the initial read won't happen, however all subsequent updates still come through. We've implemented a hack that first does a singe `read` then turns on the `sync` for that reference. This was a source of major frustration for several weeks so I'll try and comb through your code to find the problem but for now our hack suffices.
